test/integration/features/addons/addon-htpasswd-identity-provider.feature

```diff
@@ -0,0 +1,86 @@
+
+  Scenario: User can enable the htpasswd-identity-provider
+     When executing "minishift addons enable htpasswd-identity-provider" succeeds
+     Then stdout should contain "Add-on 'htpasswd-identity-provider' enabled"
+
+  Scenario: User can start Minishift with htpasswd-identity-provider add-on enabled
+    Given Minishift has state "Does Not Exist"
+      And image caching is disabled
+     When executing "minishift start" succeeds
+     Then Minishift should have state "Running"
+      And stdout should contain
+      """
+      Successfully installed addon htpasswd identity provider
+      """
+
+  Scenario: User can login to openshift instance only with default username "developer" and password "developer"
+    Given Minishift has state "Running"
+      And user waits "60" seconds
+     When executing "oc login -u developer -p developer" succeeds
+     Then stdout should contain
+      """
+      Login successful.
+      """
+     When executing "oc login -u developer -p anything" fails
+     Then stdout should contain
+     """
+     Login failed (401 Unauthorized)
+     """
+     When executing "oc login -u anything -p developer" fails
+     Then stdout should contain
+     """
+     Login failed (401 Unauthorized)
+     """
+    
+  Scenario: User can update login password for default user "developer"
+    Given Minishift has state "Running"
+     When executing "minishift addons apply htpasswd-identity-provider --addon-env USERNAME=developer --addon-env USER_PASSWORD=password" succeeds
+      And user waits "60" seconds
+     Then executing "oc login -u developer -p password" succeeds
+      And stdout should contain
+      """
+      Login successful.
+      """
+     When executing "oc login -u developer -p developer" fails
+     Then stdout should contain
+     """
+     Login failed (401 Unauthorized)
+     """
+     When executing "oc login -u developer -p anything" fails
+     Then stdout should contain
+     """
+     Login failed (401 Unauthorized)
+     """
+  
+  Scenario: User can create login credentials for openshift instance and can login using that credentials
+    Given Minishift has state "Running"
+     When executing "minishift addons apply htpasswd-identity-provider --addon-env USERNAME=openshift-user --addon-env USER_PASSWORD=openshift-password" succeeds
+      And user waits "60" seconds
+     Then executing "oc login -u openshift-user -p openshift-password" succeeds
+      And stdout should contain
+      """
+      Login successful.
+      """
+     When executing "oc login -u openshift-user -p anything" fails
+     Then stdout should contain
+     """
+     Login failed (401 Unauthorized)
+     """
+   
+  Scenario: User can use any password for default user "developer" after removing addon htpasswd-identity-provider
+    Given Minishift has state "Running"
+     When executing "minishift addons remove htpasswd-identity-provider" succeeds
+      And user waits "60" seconds
```

Comment:
everytime we use `60 seconds` of wait state, we increase the time we need to run tests...

@minishift/minishift-dev There is your win to optimize the time for CI to run, and not in cutting down on tests. Especially since the wait state is used once to 'wait' for the console/API to become responsive, but this doesn't need to be done during the rest of the test cases, as it should already be responsive...